tmk_core/common/keycode.h

```diff
@@ -521,15 +524,18 @@ enum internal_special_keycodes {
 
 enum mouse_keys {
     /* Mouse Buttons */
-    KC_MS_UP = 0xF0,
+    KC_MS_UP = 0xEC,
     KC_MS_DOWN,
     KC_MS_LEFT,
-    KC_MS_RIGHT,
+    KC_MS_RIGHT,  // 0xF0
     KC_MS_BTN1,
     KC_MS_BTN2,
     KC_MS_BTN3,
     KC_MS_BTN4,
     KC_MS_BTN5,
+    KC_MS_BTN6,
+    KC_MS_BTN7,
+    KC_MS_BTN8,
```

Comment:
You cannot shift around mouse keycodes this way — VIA now has support for mousekeys and depends on the precise values of these keycodes. Probably the only thing you can do is to add extra `KC_MS_BTN6`…`KC_MS_BTN8` keycodes starting at `0xED`, so that the existing keycodes do not change their values, and then update `IS_MOUSEKEY_BUTTON()` accordingly (and also look around for code that could assume that all `KC_MS_BTN_*` codes occupy a single continuous range, or that `KC_MS_UP` is the first mouse keycode).

Also your keycode values do not agree with comments at the moment — you are adding 3 more keycodes, so you need to start them from `0xED`, not `0xEC`.

Reply:
To be blunt, I feel that this tone is unacceptable, and violates the code of conduct.
https://github.com/qmk/qmk_firmware/blob/master/CODE_OF_CONDUCT.md

>“A gentle word turns away wrath, but a harsh word stirs up anger."

Furthermore, yes we can. We generally don't, out of respect for VIA. But VIA is NOT a QMK project and is not affiliated with QMK other than it uses QMK Firmware and has a framework built ontop of it. We have no obligation to ensure that it works. But again, out of respect, we avoid making changes that will break compatibility. 

Which is WHY this PR targets the develop branch.  This way, there is a window of time between when this is may be merged and when it hits master.  That's the entire point of the breaking changes workflow.

VIA also has a protocol definition that they could use to make sure that the correct value is used. 
https://github.com/qmk/qmk_firmware/blob/95bbd799a4f86dac37fdf2354e008d2fed7f6660/quantum/via.h#L57-L59


So while I understand the sentiment, the way you phrased it is not okay, IMO.  And fortunately, it was a QMK collaborate that the comment was targeted at, and not just somebody wanting to contribute code. 

Please try to use more friendly language. Not just because of contributors, but because this also upset me a LOT.  


That said, you are correct about the numbering. That's fixed.  
I also didn't tag wilba, because I know that he does pay attention to the PRs and has probably already seen it.